app/src/main/java/info/nightscout/androidaps/plugins/general/overview/graphData/GraphData.kt

```diff
@@ -153,20 +154,32 @@ class GraphData(
         if (useForScale) {
             maxY = 100.0 + max(overviewData.maxRatioValueFound, abs(overviewData.minRatioValueFound))
             minY = 100.0 - max(overviewData.maxRatioValueFound, abs(overviewData.minRatioValueFound))
+            overviewData.ratioScale = Scale(100.0)
             overviewData.ratioScale.setMultiplier(1.0)
-        } else
+        } else {
+            overviewData.ratioScale = Scale()
             overviewData.ratioScale.setMultiplier(maxY * scale / max(overviewData.maxRatioValueFound, abs(overviewData.minRatioValueFound)))
+        }
         addSeries(overviewData.ratioSeries)
     }
 
     // scale in % of vertical size (like 0.3)
-    fun addDeviationSlope(useForScale: Boolean, scale: Double) {
+    fun addDeviationSlope(useForScale: Boolean, scale: Double, isRatioScale: Boolean = false) {
         if (useForScale) {
             maxY = max(overviewData.maxFromMaxValueFound, overviewData.maxFromMinValueFound)
             minY = -maxY
         }
-        overviewData.dsMaxScale.setMultiplier(maxY * scale / overviewData.maxFromMaxValueFound)
-        overviewData.dsMinScale.setMultiplier(maxY * scale / overviewData.maxFromMinValueFound)
+        var graphMaxY = maxY
```

Comment:
What I mean is that `overviewData.dsMaxScale` instance is added to prepared data series in `OverviewData` before using `GraphData` for rendering. During data preparation is scale needed not known.
Then if you here assign to `overviewData.dsMinScale` diffent object, calling `setMultiplier` has no effect to scale used during  data preparation (or better say it has but only for next round, not for current data)
To apply scale immediately you must apply changes to current `Scale` object not create new

Reply:
Thanks, I'll take a look this evening to improve it.